### ISSUE TITLE
extend API version information

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#1907]
Currently galaxy can report some version information through the API, e.g:

```
{"version_major": "16.01"}
```

That's a rough estimate though, since 16.01 is obviously a moving target.
Would there be any opposition to get a bit more fine-grained information?
Perhaps a `commit.xml.sample` config file, which would be automatically annotated by travis upon merging?
In addition, it would be nice to have a user-definable tag.
Ultimately, I think it would be cool if a call to /api/version would return something like

```
{"version_major":"16.04", `"commit-id":"abcde123", "user_tag":"_fix_job_runner_issue"}
```

I think this is especially important when not cloning through git, and you can't just do a  `git branch` or `git log`. The user_tag would be nice, since one could pass that along into log tracking systems, to identify problems when changing more general stuff that is not directly related to the galaxy code base.

**Comment by hexylena** [comment]
:+1: from me. I was upgrading some of our internal apps to use sentry and noticed they too [track the precise version](https://docs.getsentry.com/hosted/clients/python/integrations/django/) which thry.call 'release'. That sort of extra info would be great

**Comment by mvdbeek** [comment]
yes, that's primarily what I had in mind, but also this would be nice for issues like this:
https://github.com/bgruening/docker-galaxy-stable/issues/152.
Also it could accelerate bug fixing on the mailing list if exceptions and errors would rigorously include version and commit-ids.

**Comment by hexylena** [comment]
xref #10937
